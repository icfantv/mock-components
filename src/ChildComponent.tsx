import * as React from 'react'
import { BindAll } from 'lodash-decorators'

@BindAll()
export class ChildComponent extends React.Component {
  public render(): React.ReactElement<ChildComponent> {
    return <div>Child Component</div>
  }
}
