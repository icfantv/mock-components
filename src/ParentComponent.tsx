import * as React from 'react'
import { BindAll } from 'lodash-decorators'
import { ChildComponent } from 'ChildComponent'

@BindAll()
export class ParentComponent extends React.Component {
  public render(): React.ReactElement<ParentComponent> {
    return (
      <div>
        <div>Parent Component</div>
        <div>
          <ChildComponent />
        </div>
      </div>
    )
  }
}
