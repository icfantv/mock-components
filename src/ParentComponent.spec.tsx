import * as React from 'react'
import { create } from 'react-test-renderer'
import { cleanup, render } from 'react-testing-library'

import { ChildComponent } from 'ChildComponent'
import { ParentComponent } from 'ParentComponent'
import { stubComponent } from 'test/utils'

describe('test the <ParentComponent/> component', () => {
  describe('using react-test-renderer', () => {
    test('it should deep render the parent component', () => {
      const tree = create(<ParentComponent />).toJSON()
      expect(tree).toMatchSnapshot()
    })

    describe('test the <ParentComponent/> component with stubbing', () => {
      stubComponent(ChildComponent)
      test('it should shallow render the parent component', () => {
        const tree = create(<ParentComponent />).toJSON()
        expect(tree).toMatchSnapshot()
      })
    })

    test('it should deep render the parent component again', () => {
      const tree = create(<ParentComponent />).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })

  describe('using react-testing-library', () => {
    afterEach(cleanup)

    test('it should deep render the parent component', () => {
      const { container } = render(<ParentComponent />)
      expect(container.firstChild).toMatchSnapshot()
    })

    describe('test the <ParentComponent/> component with stubbing', () => {
      stubComponent(ChildComponent)

      test('it should shallow render the parent component again', () => {
        const { container } = render(<ParentComponent />)
        expect(container.firstChild).toMatchSnapshot()
      })
    })
  })
})
