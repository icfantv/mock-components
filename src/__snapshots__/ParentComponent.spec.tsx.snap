// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test the <ParentComponent/> component it should deep render the parent component 1`] = `
<div>
  <div>
    Parent Component
  </div>
  <div>
    <div>
      Child Component
    </div>
  </div>
</div>
`;

exports[`test the <ParentComponent/> component it should deep render the parent component again 1`] = `
<div>
  <div>
    Parent Component
  </div>
  <div>
    <div>
      Child Component
    </div>
  </div>
</div>
`;

exports[`test the <ParentComponent/> component test the <ParentComponent/> component with stubbing it should shallow render the parent component 1`] = `
<div>
  <div>
    Parent Component
  </div>
  <div />
</div>
`;
